yjit/src/asm/mod.rs

```diff
@@ -87,6 +87,9 @@ pub struct CodeBlock
     // References to labels
     label_refs: Vec<LabelRef>,
 
+    // Comments for assembly instructions, if that feature is enabled
+    asm_comments: Vec<(usize, String)>,
```

Comment:
Can probably switch to a `BTreeMap<usize, String>` and get dedup and sorting for log time.

Reply:
The dedup is currently best effort since sometimes we rewind the write cursor back to the middle of the codeblock to patch the code, which defeats checking the last element. For `RubyVM::YJIT.disasm` we also query for comments in a range, which is linear on the number of total comments at the moment.

Reply:
Alan's idea of using a BTreeMap seems like a good one